src/app/components/home/components/ip-summary/ip-summary.component.ts

```diff
@@ -94,6 +99,27 @@ export class IPSummaryComponent implements OnChanges {
     });
   }
 
+  getBlacklist() {
+    const response = this.service.blacklist();
+
+    response.subscribe({
+      next: (data: string) => {
+        this.blacklist = data.split('\n').filter(ip => ip.trim() !== '');
+      },
+      error: (err) => { 
+        console.log(err);
+
+        this.summary = {} as SummaryObj;
+
+        this.error = {
+          val: true,
+          msg: err.message,
+          code: err.status,
+        };
+       },
+    });
+  }
+
```

Comment:
Pode ver se isso funciona? Creio que usando as funções `pipe` e `map` do RxJS seja mais declarativo e funcional, talvez o professor prefira assim.
```suggestion
  import { map } from 'rxjs';

  getBlacklist() {
    this.service
      .blacklist()
      .pipe(map((data: string) => data.split('\n').filter((ip) => ip.trim() !== '')))
      .subscribe({
        next: (blacklist: string[]) => {
          this.blacklist = blacklist;
        },
        error: (err) => {
          console.log(err);

          this.summary = {} as SummaryObj;

          this.error = {
            val: true,
            msg: err.message,
            code: err.status,
          };
        },
      });
  }
```

Reply:
Testei a apliquei a sugestão